code/modules/mob/living/simple_animal/hostile/terror_spiders/objects.dm

```diff
@@ -0,0 +1,341 @@
+// ----------------- TERROR SPIDERS: EGGS (USED BY NURSE AND QUEEN TYPES) ---------
+// --------------------------------------------------------------------------------
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoLayTerrorEggs(var/lay_type, var/lay_number, var/lay_crawl)
+	stop_automated_movement = 1
+	var/obj/effect/spider/terror_eggcluster/C = new /obj/effect/spider/terror_eggcluster(get_turf(src))
+	C.spiderling_type = lay_type
+	C.spiderling_number = lay_number
+	C.spiderling_ventcrawl = lay_crawl
+	C.faction = faction
+	C.spider_myqueen = spider_myqueen
+	C.master_commander = master_commander
+	C.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+	C.enemies = enemies
+	//In future, might use something like: C.name = initial(lay_type.egg_name)
+	//However, right now, that doesn't work.
+	if (spider_growinstantly)
+		C.amount_grown = 250
+		C.spider_growinstantly = 1
+	spawn(10)
+		stop_automated_movement = 0
+	if (spider_queen_declared_war)
+		C.spider_queen_declared_war = 1
+
+
+/obj/effect/spider/terror_eggcluster
+	name = "terror egg cluster"
+	desc = "A cluster of tiny spider eggs. They pulse with a strong inner life, and appear to have sharp thorns on the sides."
+	icon_state = "eggs"
+	var/amount_grown = 0
+	var/spider_growinstantly = 0
+	var/spider_queen_declared_war = 0 // set to 1 by procs
+	var/faction = list("terrorspiders")
+	var/spider_myqueen = null
+	var/master_commander = null
+	var/spiderling_type = null
+	var/spiderling_number = 1
+	var/spiderling_ventcrawl = 1
+	var/ai_playercontrol_allowingeneral = 1
+	var/list/enemies = list()
+
+
+/obj/effect/spider/terror_eggcluster/New()
+	pixel_x = rand(3,-3)
+	pixel_y = rand(3,-3)
+	processing_objects.Add(src)
+	// Not a mistake that this does not call ..().
+	// /obj/effect/spider/stickyweb/New() does not call ..() either.
+
+
+/obj/effect/spider/terror_eggcluster/process()
+	amount_grown += rand(0,2)
+	if(amount_grown >= 100)
+		var/num = spiderling_number
+		for(var/i=0, i<num, i++)
+			var/obj/effect/spider/terror_spiderling/S = new /obj/effect/spider/terror_spiderling(get_turf(src))
+			if (spiderling_type)
+				S.grow_as = spiderling_type
+			if (spiderling_ventcrawl)
+				S.use_vents = spiderling_ventcrawl
+			S.faction = faction
+			S.spider_myqueen = spider_myqueen
+			S.master_commander = master_commander
+			S.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+			S.enemies = enemies
+			if (spider_queen_declared_war)
+				S.spider_queen_declared_war = 1
+			if (spider_growinstantly)
+				S.amount_grown = 250
+		var/rnum = 5 - spiderling_number
+		for(var/i=0, i<rnum, i++)
+			var/obj/effect/spider/terror_spiderling/S = new /obj/effect/spider/terror_spiderling(get_turf(src))
+			S.stillborn = 1
+			// the idea is that every set of eggs always spawn 5 spiderlings, but most are not going to grow up, just some do.
+		qdel(src)
+
+
+/obj/effect/spider/terror_spiderling
```

Comment:
90% of the code for this is in process(), and it is different from the normal spiderling's process(). The differences are not conducive to simply doing ..() and putting something before or after it, either - they're things like changing the text in the middle of the proc. I don't think that making it a child would reduce code length significantly. In addition, when these were children of regular spider entities back in the day, that introduced a lot of bugs with things inherited from them that I did not want. Maybe you could replace die() and 3 lines from New() with ..() if it was a child of regular spiderling, but it wouldn't shrink process(), which is the lion's share of the code.